### PR TITLE
Create contracts with cross-contract call

### LINKED ISSUE #1: Get cross-contract calls working!
Nothing works, I'm pretty sure. But we can start out by reading the README and undestanding what's supposed to happen.

Also, this should help, I drew it myself:
PXL_20220822_072501530 MP_2

Then, immerse yourself in love, and in this entire page. Memorize it:
https://docs.cosmwasm.com/docs/1.0/smart-contracts/contract-semantics/#submessages

### PR DESCRIPTION
**Pull request by nlipartiia-hacken**
Closes #1
Changed `dinner` and `demo-totals` so that they work and `demo-totals` makes a cross-contract call to `dinner`.
Created tests and scripts `build.sh` and `deploy.sh` for building and deploying to uni-3 testnet.

Also `dinner` doesn't actually make cross-contract call to `cw1_whitelist` through submessages, it only queries the state:
```
deps.querier.query_wasm_smart(
    SCHOLARSHIPS_ADDRESS.load(deps.storage)?,
    &cw1_whitelist::msg::QueryMsg::AdminList::<String> {},
)?;
```
Is it okay for this example?

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (mikedotexe) [inline deploy.sh:1]: I think this is 644 (not executable) and might need to be changed, like with:

    chmod +ux deploy.sh

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (mikedotexe) [inline build.sh:1]: I think this one also needs

    chmod +ux build.sh

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+RES_DINNER=$(junod tx wasm store artifacts/cross_contract_dinner.wasm --from $OWNER $TXFLAG -y --output json -b block)
+CODE_DINNER=$(echo $RES_DINNER | jq -r '.logs[0].events[-1].attributes[0].value')
+
+RES_WHITELIST=$(junod tx wasm store contracts/whitelist-binary/cw1_whitelist.wasm --from $OWNER $TXFLAG -y --output json -b block)
+CODE_WHITELIST=$(echo $RES_WHITELIST | jq -r '.logs[0].events[-1].attributes[0].value')
+
+# Instantiate
+INIT_WHITELIST='{"admins":["'$(junod keys show -a $ALICE)'"], "mutable": true}'
+junod tx wasm instantiate $CODE_WHITELIST "$INIT_WHITELIST" --from $OWNER --label "demo-total" $TXFLAG -y --no-admin
```
Comment (mikedotexe) [inline deploy.sh]:
```suggestion
junod tx wasm instantiate $CODE_WHITELIST "$INIT_WHITELIST" --from $OWNER --label "scholarship-list" $TXFLAG -y --no-admin
```

File: deploy.sh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+CONTRACT_WHITELIST=$(junod query wasm list-contract-by-code $CODE_WHITELIST $NODE --output json | jq -r '.contracts[-1]')
+
+INIT_DEMO_TOTAL='{"denom":"ujunox"}'
+junod tx wasm instantiate $CODE_DEMO_TOTAL "$INIT_DEMO_TOTAL" --from $OWNER --label "demo-total" $TXFLAG -y --no-admin
+CONTRACT_DEMO_TOTAL=$(junod query wasm list-contract-by-code $CODE_DEMO_TOTAL $NODE --output json | jq -r '.contracts[-1]')
+
+INIT_DINNER='{"denom":"ujunox", "scholarship_address":"'$CONTRACT_WHITELIST'"}'
+junod tx wasm instantiate $CODE_DINNER "$INIT_DINNER" --from $OWNER --label "demo-total" $TXFLAG -y --no-admin
```
Comment (mikedotexe) [inline deploy.sh]:
```suggestion
junod tx wasm instantiate $CODE_DINNER "$INIT_DINNER" --from $OWNER --label "dinner" $TXFLAG -y --no-admin
```

**Review by mikedotexe — CHANGES_REQUESTED** [review]

**Comment by mikedotexe** [comment]
These look like they're working great))

### non-scholarship person can't register
    
    junod tx wasm execute juno1sn3fzqc98mzy7jpg5v478nh6s6r5wnc7fygwsnlraeez6skvl4wsrt64ct '{"register_with_scholarship":{"dinner_contract":"juno1upxer95zx4l6xrlju7pd8t02fj788m0t9x8seqkt0khh70qnrt4sxypucs"}}' --from cross-contract-example-owner --node https://rpc.uni.juno.deuslabs.fi:443 --chain-id uni-3 --gas-prices 0.025ujunox --gas auto --gas-adjustment 1.3 --broadcast-mode block -y

### no payment can't register
    
    junod tx wasm execute juno1sn3fzqc98mzy7jpg5v478nh6s6r5wnc7fygwsnlraeez6skvl4wsrt64ct '{"register_with_payment":{"dinner_contract":"juno1upxer95zx4l6xrlju7pd8t02fj788m0t9x8seqkt0khh70qnrt4sxypucs"}}' --from cross-contract-example-owner --node https://rpc.uni.juno.deuslabs.fi:443 --chain-id uni-3 --gas-prices 0.025ujunox --gas auto --gas-adjustment 1.3 --broadcast-mode block -y

**Comment by mikedotexe** [comment]
Some testnet info that may be helpful:

totals (3309)
juno1sn3fzqc98mzy7jpg5v478nh6s6r5wnc7fygwsnlraeez6skvl4wsrt64ct

    junod q wasm contract-state smart juno1sn3fzqc98mzy7jpg5v478nh6s6r5wnc7fygwsnlraeez6skvl4wsrt64ct '{"registrant_number":{}}' --node https://rpc.uni.juno.deuslabs.fi:443 --output json

dinner (3310)
juno1upxer95zx4l6xrlju7pd8t02fj788m0t9x8seqkt0khh70qnrt4sxypucs

    junod q wasm contract-state smart juno1upxer95zx4l6xrlju7pd8t02fj788m0t9x8seqkt0khh70qnrt4sxypucs '{"get_all_registrants":{}}' --node https://rpc.uni.juno.deuslabs.fi:443 --output json

whitelist (3311)
juno1xfyru5t2h6gry008n78ju58r5gp4upkxhl7l6adj56850x7u4rsqxmuse6

    junod q wasm contract-state smart juno1xfyru5t2h6gry008n78ju58r5gp4upkxhl7l6adj56850x7u4rsqxmuse6 '{"admin_list":{}}' --node https://rpc.uni.juno.deuslabs.fi:443 --output json

File: deploy.sh (hunk 1 above)
Comment (nlipartiia-hacken) [reply deploy.sh:1]: Oops, yes. Now it's 755

**Review by mikedotexe — APPROVED** [review]

**Pull request merged**